### ISSUE TITLE
where is client B in VoipPhone params?

### ISSUE THREAD

**Issue by rushant001** [tayler6000/pyVoIP#3]
phone=VoIPPhone(<SIP Server IP>, <SIP Server Port>, <SIP Server Username>, <SIP Server Password>, callCallback=answer, myIP=<Your computer's local IP>, sipPort=<Port to use for SIP (int, default 5060)>, rtpPortLow=<low end of the RTP Port Range>, rtpPortHigh=<high end of the RTP Port Range>)
where is client B params in 'VoIPPhone' ?

![image](https://user-images.githubusercontent.com/21120232/144593587-fe316d28-0988-4826-91bc-20ff3f68c32f.png)

**Comment by tayler6000** [comment]
`VoIPPhone()` is a client.  If you wanted, you could make a second `VoIPPhone()` instance as client B. Client B could also be a hard phone or softphone.

I believe you're asking how to call client B, as shown in this diagram.  As I said, `VoIPPhone()` is a client, and the parameters needed for initialization are only used for connecting to the SIP server.  To initiate a phone call, as shown in this diagram, you would use the following code.  Assume `clientA` is an instance of `VoIPPhone()`.

```python
clientA.call(clientB_Phone_Number)
```

I hope that helps!   If you have any follow-up questions, please post an additional comment on this thread!

**Comment by rushant001** [comment]
Thanks for your reply sincerely.I'll try it later.
Do you plane to add some test code as samples please?

**Comment by tayler6000** [comment]
I do have some example code on the [docs page](https://pyvoip.readthedocs.io/en/latest/Examples.html).

**Comment by rushant001** [comment]
![image](https://user-images.githubusercontent.com/21120232/144693362-7ebda90c-0355-4e88-b8a9-a74776b078bb.png)

As shown in the above example, how could i make an Announcement on client B?

**Comment by tayler6000** [comment]
That example is for a client you would call to get your announcements.  Are you wanting code for something (client A) that would automatically call another client (client B) when there is a new announcement?

**Comment by rushant001** [comment]
> That example is for a client you would call to get your announcements. Are you wanting code for something (client A) that would automatically call another client (client B) when there is a new announcement?

Yes.How to send an announcement from client A to Client B via sip server?

**Comment by tayler6000** [comment]
This is a refactored version of some other code I was working on for a similar effect.  I haven't run this specific version so there may be a few errors, but this should work.

```python
from pyVoIP.VoIP import VoIPPhone, InvalidStateError, CallState
import threading
import time

SUBSCRIBED = [clientB_Phone_Number, clientC_Phone_Number]

def makeCall(number, announcement):
  global clientA
  call = clientA.call(number)
  while call.state == CallState.DAILING:
    time.sleep(.1) # Wait until the call is answered.
  time.sleep(1) # Wait until call is properly established.
  if call.state == CallState.ANSWERED:
    call.writeAudio(announcement)
    length = len(announcement) / 8000
    stop = time.time() + length
    while call.state == CallState.ANSWERED and time.time() <= stop:
      time.sleep(.1)
    time.sleep(1)
    try:
      call.hangup()
    except:
      pass

def answer(call):
  '''
  When clientA is called, it will immediately start recording the line until 1 is pressed by the caller.
  
  Once the caller has pressed 1, it will distribute the announcement to other clients.
  '''
  try:
    call.answer()
    
    announcement = b""
    dtmf = ""
    
    while dtmf != '1' and call.state == CallState.ANSWERED:
      announcement += call.readAudio()
      dtmf = call.getDTMF()
    call.hangup()
  except InvalidStateError:
    pass
  except:
    call.hangup()
  
  global SUBSCRIBED
  for number in SUBSCRIBED:
    t = threading.Timer(1, makeCall, [number, announcement])
    t.start()
    
if __name__ == '__main__':
  clientA=VoIPPhone(<SIP Server IP>, <SIP Server Port>, <SIP Server Username>, <SIP Server Password>, callCallback=answer, myIP=<Your computers local IP>)
  clientA.start()
  input('Press enter to disable the phone')
  clientA.stop()
```
